rfcs/button-group-component.md

```diff
@@ -0,0 +1,43 @@
+# Button Group Component
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Button} from 'baseui/button';
+import {ButtonGroup} from 'baseui/button-group';
+
+export default () => {
+  return
+    <ButtonGroup>
```

Comment:
button group can change styles around button, so we need a root component to be exposed for that

Reply:
did we add extra root element to be able to change padding\margin, etc around buttons?